hoot-core/src/main/cpp/hoot/core/ops/UnconnectedWaySnapper.cpp

```diff
@@ -561,73 +559,118 @@ bool UnconnectedWaySnapper::_snapUnconnectedNodeToWay(NodePtr nodeToSnap)
-      const geos::geom::Coordinate closestWayToSnapToCoord =
-        OsmUtils::closestWayCoordToNode(nodeToSnap, wayToSnapTo,
-          shortestDistanceFromNodeToSnapToWayCoord, _snapToWayDiscretizationSpacing, _map);
-
-      // This check of the calculated distance being less than the allowed snap distance should not
-      // be necessary, but it is for now.  For some reason, neighbors are occasionally being
-      // returned at longer distances away than expected.
-      if (/*shortestDistance != DBL_MAX &&*/
-          shortestDistanceFromNodeToSnapToWayCoord <= _maxSnapDistance)
-      { 
-        // snap the node to the way
-
-        // figure out where on the target way to insert our node being snapped
-        const int nodeToSnapInsertIndex = _getNodeToSnapWayInsertIndex(nodeToSnap, wayToSnapTo);
-
-        if (Log::getInstance().getLevel() <= Log::Debug)
-        {
-          // This projection could be expensive, so do it during debugging only.
-          const geos::geom::Coordinate wgs84Coord =
-            MapProjector::project(
-              closestWayToSnapToCoord, _map->getProjection(),
-              MapProjector::createWgs84Projection());
-          LOG_TRACE(
-            "Snapping way node: " << nodeToSnap->getId() << " to coord: " <<
-            closestWayToSnapToCoord.toString() << " (wgs84: " << wgs84Coord.toString() <<
-            ") and inserting at index: " << nodeToSnapInsertIndex);
-        }
+      _snappedWayNodeIds.append(nodeToSnap->getId());
+      _numAffected++;
+      _numSnappedToWays++;
 
-        // move the snapped node to the closest way coord
-        nodeToSnap->setX(closestWayToSnapToCoord.x);
-        nodeToSnap->setY(closestWayToSnapToCoord.y);
+      // Don't snap the node more than once.
+      return true;
+    }
+  }
 
-        // Add the optional custom tag for tracking purposes.
-        if (_markSnappedNodes)
-        {
-          nodeToSnap->getTags().set(MetadataTags::HootSnappedWayNode(), "snapped_to_way");
-        }
+  return false;
+}
 
-        // add the snapped node as a way node on the target way
-        QList<long> wayNodeIdsToSnapToList =
-          QList<long>::fromVector(QVector<long>::fromStdVector(wayNodeIdsToSnapTo));
-        wayNodeIdsToSnapToList.insert(nodeToSnapInsertIndex, nodeToSnap->getId());
-        wayToSnapTo->setNodes(wayNodeIdsToSnapToList.toVector().toStdVector());
-        LOG_VART(wayToSnapTo->getNodeIds());
+bool UnconnectedWaySnapper::snapClosestEndpointToWay(OsmMapPtr map, WayPtr disconnected, WayPtr connectTo)
+{
+  //  Create object for static call
+  UnconnectedWaySnapper uws;
+  uws._map = map;
+  //  Call protected function to snap closest endpoint to way
+  return uws._snapClosestEndpointToWay(disconnected, connectTo);
+}
 
-        _snappedWayNodeIds.append(nodeToSnap->getId());
-        _numAffected++;
-        _numSnappedToWays++;
+bool UnconnectedWaySnapper::_snapClosestEndpointToWay(WayPtr disconnected, WayPtr connectTo)
```

Comment:
Finds the endpoint node nearest the `connectTo` way and then calls the function to do the actual connecting.